aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -79,7 +79,7 @@ int64_t SparseTensorImpl::storage_offset() const {
   AT_ERROR("sparse tensors do not have storage");
 }
 void SparseTensorImpl::set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values) {
-  AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe is not allowed on Tensor created from .data or .detach()");
+  AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe ", err_msg_for_changing_metadata_on_derived_tensor);
```

Comment:
I don't think you should start with the function name; the issue here is that you start the message with something that the user knows nothing about (what's set_indices_and_values_unsafe, that's not exposed anywhere in the normal Python or C++ API).